### PR TITLE
Implement actions and events.

### PR DESCRIPTION
**Pull request by mrstegeman**

### CODE REVIEW

File: lib/device.js [hunk 1]
```diff
@@ -22,37 +22,65 @@ class Device {
...
   asDict() {
-    var properties = {};
+    const properties = {};
     this.properties.forEach((property, propertyName) => {
       properties[propertyName] = property.asDict();
     });
+
+    const actions = {};
+    this.actions.forEach((metadata, actionName) => {
+      actions[actionName] = Object.assign({}, metadata);
```
Comment (hobinjk) [inline lib/device.js]: Why isn't this asDict?

File: lib/action.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+      timeCompleted: this.timeCompleted,
+    };
+  }
+
+  /**
+   * Get this action's ID.
+   *
+   * @returns {String} The ID.
+   */
+  getId() {
```
Comment (hobinjk) [inline lib/action.js]: Why have all of these unused getters for properties?

File: lib/event.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      timestamp: this.timestamp,
+    };
+  }
+
+  /**
+   * Get the device associated with this event.
+   *
+   * @returns {Object} The device.
+   */
+  getDevice() {
```
Comment (hobinjk) [inline lib/event.js]: Same as action, these unused getters seem extraneous.

File: lib/action.js (hunk 2 above)
Comment (mrstegeman) [reply lib/action.js]: I copied action.js and event.js from webthing-node. I can remove them.

File: lib/device.js (hunk 1 above)
Comment (mrstegeman) [reply lib/device.js]: It should be.

**Pull request merged**